.github/workflows/cypress-test-oidc-e2e.yml

```diff
@@ -0,0 +1,192 @@
+      - name: Sleep while OpenSearch starts
+        uses: peternied/action-sleep@v1
+        with:
+          seconds: 30
+
+      # Verify that the server is operational
+      - name: Check OpenSearch Running on Linux
+        if: ${{ runner.os != 'Windows'}}
+        run: curl https://localhost:9200/_cat/plugins -u 'admin:admin' -k -v
+        shell: bash
+
+      - if: always()
+        run: cat ./opensearch-${{ env.OPENSEARCH_VERSION }}-SNAPSHOT/logs/opensearch.log
+        shell: bash
+
+      # OSD bootstrap
+      - name: Run Dashboard with Security Dashboards Plugin
+        uses: ./.github/actions/install-dashboards
+        with:
+          plugin_name: security-dashboards-plugin
+
+      # Configure the Dashboard for OIDC setup
+      - name: Configure and Run OpenSearch Dashboards with Cypress Test Cases
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          cd ./OpenSearch-Dashboards
+          echo 'server.host: "localhost"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.hosts: ["https://localhost:9200"]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.ssl.verificationMode: none' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.username: "kibanaserver"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.password: "kibanaserver"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.requestHeadersWhitelist: [ authorization,securitytenant ]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.multitenancy.enabled: true' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.multitenancy.tenants.preferred: ["Private", "Global"]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.readonly_mode.roles: ["kibana_read_only"]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.cookie.secure: false' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.openid.connect_url: "http://127.0.0.1:8080/auth/realms/master/.well-known/openid-configuration"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.openid.client_id: "opensearch"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.openid.client_secret: "${{ env.TEST_KEYCLOAK_CLIENT_SECRET }}"'>> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.auth.type: ["openid"]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.auth.multiple_auth_enabled: true' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.ui.openid.login.buttonname: "OIDC"' >> ./config/opensearch_dashboards.yml
+          echo 'home.disableWelcomeScreen: true' >> ./config/opensearch_dashboards.yml
+          echo 'HERE IS THE DASHBOARD CONFIG'
+          cat ./config/opensearch_dashboards.yml
+          nohup yarn start --no-base-path --no-watch &
+          sleep 600
```

Comment:
Can we wait on something specific, e.g. loop{ curl;  sleep;}, to know the tests can execute ?

Reply:
Here is the solution: By using a loop block, it was looking for the phrase `bundles compiled successfully after` in the sever log. The loop will keep checking the log until it finds this msg, and proceed to the next step immediately. It also has a max timeout for 600 seconds. If the timeout gets reached, it will exit.

Successful output is like this: https://github.com/opensearch-project/security-dashboards-plugin/actions/runs/7028407676/job/19124329423?pr=1579#step:19:103